pkg/store/store_kvdb.go

```diff
@@ -129,6 +135,11 @@ func (k *kvdbPersistenceStore) Set(
 	if err != nil {
 		if err == kvdb.ErrExist {
 			return fmt.Errorf("secret with name %v already exists.", secretId)
+		} else if err == kvdb.ErrExist && override {
```

Comment:
We should check this before the first `if` condition else it will never come here.